assets/controllers/oney/popin_controller.js

```diff
@@ -0,0 +1,164 @@
+        }, 500)
+      );
+    }
+  }
+  handleProductOptionsChange($selectors, prop) {
+    if (prop === "option") {
+      let selector = "";
+      $selectors.each((k, v) => {
+        const select = $(v);
+        const option = select.find("option:selected").val();
+        selector += `[data-${select.attr("data-option")}="${option}"]`;
+      });
+      return (this.productMetaValue = {
+        ...this.productMetaValue,
+        product_variant_code: $(this.variantCodesTarget).find(selector).attr("data-value")
+      });
+    }
+  }
+  check() {
+    this.storage = [];
+    this.toggleLoader();
+    $.ajax({
+      url: this.eligibleUrlValue,
+      data: this.productMetaValue,
+      success: (res) => {
+        $(this.element).find(".oney-logo").attr("src", this.productMetaValue.img[res.isEligible]);
+        res.isEligible
+          ? $(this.popinTarget).removeClass("disabled").addClass("enabled")
+          : $(this.popinTarget).removeClass("enabled").addClass("disabled");
+      },
+      complete: () => {
+        this.toggleLoader();
+      }
+    });
+  }
+  /**
+   * https://davidwalsh.name/javascript-debounce-function
+   */
+  debounce(func, wait) {
+    let timeout;
+    return function executedFunction(...args) {
+      const later = () => {
+        clearTimeout(timeout);
+        func(...args);
+      };
+      clearTimeout(timeout);
+      timeout = setTimeout(later, wait);
+    };
+  }
+  fade() {
+    $(this.element).on("click", (e) => {
+      e.preventDefault();
+      e.stopPropagation();
+      if (!$(this.popinTarget).is(":empty") && $(this.popinTarget).text().trim() === this.storage) {
+        $(this.popinTarget).fadeIn();
+        return;
+      }
+      // content isn't loaded yet
+      this.toggleLoader();
+      this.load();
+    });
+  }
+  load() {
+    $.ajax({
+      url: this.popinUrlValue,
+      data: this.productMetaValue,
+      success: (res) => {
+        if (res.includes(this.translationsValue.reason)) {
+          $(this.popinTarget).removeClass("enabled").addClass("disabled");
+        }
+        this.storage = $(res).text().trim();
+        $(this.popinTarget).html(res);
+      },
+      error: ()=>  {
+        $(this.popinTarget).removeClass("enabled").addClass("disabled").html(`
+          <div class="oney-popin__header">
+            <a class="close" href="javascript:void(0);" title="${this.translationsValue.close}">
+              <span></span><span></span>
+            </a>
+          </div>
+          <div class="oney-popin__content">
+            <p class="reasons">${this.translationsValue.reason}</p>
+          </div>
+        `);
+      },
+      complete: ()=>  {
+        this.toggleLoader();
+        $(this.popinTarget).fadeIn();
+        this.closeHandler();
+      },
+    });
+  }
+  toggleLoader() {
+    $(this.element).find('.sylius-shop-loader').toggleClass("d-none");
+  }
+  closeHandler() {
+    $("html")
+      .not(this.popinTarget)
+      .on("click", (e) => {
```

Comment:
Is there any "once" or something like this to avoid registering multiple event listener on html ?

Reply:
yes https://api.jquery.com/one/ i guess